### PR TITLE
Feature: Redesign diagram list

### LINKED ISSUE #278: Project page (index page): redesign diagrams list
## Description

On the project page (index page), redesign the diagrams actual grid to fit to the mockup.

To implements:
- List of diagrams with these columns:
  - Category (tag of plugins)
  - Diagram name with only the complete path of diagrams
  - plugin icon
  - action menu with only modify and delete option. Modify option is rename option.

Not to implements:
  - column: Link, environment, Compliance status and "{xxxxxxx}" :interrobang: 
  - the buttons to switch grid and list above the search bar
  - Global panel style
  - search bar
  - Project summary
  - "Crystal infra"
  - Keep template grid below the diagrams list

## Mockup

image

### PR DESCRIPTION
**Pull request by justinehell**
Resolve #278

### CODE REVIEW

File: src/components/grid/DiagramGrid.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<template>
+  <div
+    v-if="diagrams?.length === 0"
+    class="row text-center text-h6 text-grey empty-grid"
```
Comment (Zorin95670) [inline src/components/grid/DiagramGrid.vue:4]: missing class name `diagram-grid`

File: src/components/table/DiagramTable.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
+<template>
+  <q-table
```
Comment (Zorin95670) [inline src/components/table/DiagramTable.vue:2]: missing class name `diagram-table`

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/components/grid/DiagramGrid.vue (hunk 1 above)
Comment (justinehell) [reply src/components/grid/DiagramGrid.vue:4]: done

File: src/components/table/DiagramTable.vue (hunk 2 above)
Comment (justinehell) [reply src/components/table/DiagramTable.vue:2]: done

**Review by Zorin95670 — APPROVED** [review]

File: tests/unit/components/ModelsView.spec.js [hunk 3]
```diff
@@ -82,14 +82,14 @@ describe('Test component: ModelsView', () => {
...
-    it('should call router.push with the given model parameters', () => {
+  describe('Test function: onDiagramClick', () => {
+    it('should call router.push with the given diagram parameters', () => {
       const model = {
         plugin: 'pluginName',
         name: 'modelName',
       };
 
-      wrapper.vm.onModelCardClick(model);
+      wrapper.vm.onDiagramClick(model);
```
Comment (k-y0u) [inline tests/unit/components/ModelsView.spec.js:92]: model -> diagram

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/PluginManager.js [hunk 4]
```diff
@@ -111,12 +111,31 @@ export function getPlugins() {
  * Get instantiated plugin corresponding to the given name.
  *
  * @param {String} name - Name of the plugin to retrieve.
- * @return {Object} Return an array of plugin.
+ * @return {Object} Return a plugin.
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/PluginManager.js:114]: done

File: src/composables/PluginManager.js [hunk 5]
```diff
@@ -111,12 +111,31 @@ export function getPlugins() {
...
+ * @return {Object} Return an array of tags otherwise an empty array.
+ */
+export function getPluginTags(name) {
+  return getPluginByName(name)?.configuration.tags || [];
+}
+
+/**
+ * Get all tags from all plugins.
+ *
+ * @return {Object} Return an array of plugin.
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/PluginManager.js:133]: done

File: src/composables/Project.js [hunk 6]
```diff
@@ -730,7 +730,12 @@ export async function getPluginModels(modelsdefaultFolder, pluginName) {
   return Promise.allSettled(dirEntries.map(
     (entry) => (async () => {
       const isDir = await isDirectory(`${modelsdefaultFolder}/${pluginName}/${entry}`);
-      return isDir ? { name: entry, plugin: pluginName } : null;
+      return isDir ? {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/Project.js:733]: done

File: tests/unit/components/ModelsView.spec.js (hunk 3 above)
Comment (justinehell) [reply tests/unit/components/ModelsView.spec.js:92]: done

**Review by k-y0u — DISMISSED** [review]

**Review by Zorin95670 — DISMISSED** [review]

**Review by k-y0u — DISMISSED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**